### DISCUSSION TITLE
How to use the Largest Connected Component transform?

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#6941] [Q&A] [answered]
Hi! @rusty1s @EdisonLeeeee

Sorry if this is a dumb question. I am trying to use the LCC transform for node classification. Is this the right way to use it?

```python
newdata = LargestConnectedComponents(data)
print(newdata)
LargestConnectedComponents(Data(x=[2708, 1433], edge_index=[2, 10556], y=[2708], train_mask=[2708], val_mask=[2708], test_mask=[2708]))
```

How do I access the data attributes?
Im getting an error -
```python
print(newdata.x) 
AttributeError: 'LargestConnectedComponents' object has no attribute 'x'
```
I tried converting to NetworkX choosing the LCC and then converting back to PyG graph. But the y-labels still show the original number instead of the LCC number.

Thanks!

**Comment by AdarshMJ** [comment, marked as answer]
Sorry my bad. I got it. Thanks again for this wonderful library!

**Reply by mlv42** [reply]
Hi @AdarshMJ !
Could you share how to use it?

Thanks a lot!

**Reply by AdarshMJ** [reply]
Hi you can use it like this - 

```Python
dataset = Planetoid(root=path, name=datasetname,transform=NormalizeFeatures())
data = dataset[0]
print(f"Selecting the LargestConnectedComponent..")
transform = LargestConnectedComponents()
data = transform(dataset[0])
```

**Reply by rusty1s** [reply]
Yes, this looks correct.
